akka-stream-tests/src/test/java/akka/stream/javadsl/FlowTest.java

```diff
@@ -382,6 +383,55 @@ public void mustBeAbleToUseMerge() throws Exception {
   }
 
   @Test
+  public void mustBeAbleToUseAsyncMerge() throws Exception {
```

Comment:
If you look at the other test methods in this test, they all mention what stage they test, while this doesn't. `mustBeAbleToUsefromSourceCompletionStage` would be good to me and follow the style of the rest of the test. 

However, if this is indeed is the thing it tests (which was unclear and that was the reason I asked) then that already has test coverage in your stage scala test, as far as I can see this does not add anything to that and can be removed.